samples/file-encryptor/enclave/openssl/keys.cpp

```diff
@@ -0,0 +1,428 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openssl/rand.h>
+#include <string.h>
+#include "../common/common.h"
+#include "encryptor.h"
+
+#define ENCRYPT_OPERATION true
+#define DECRYPT_OPERATION false
+#define SALT_SIZE_IN_BYTES 16 // Length of salt size
+
+void ecall_dispatcher::dump_data(
+    const char* name,
+    unsigned char* data,
+    size_t data_size)
+{
+    TRACE_ENCLAVE("Data name: %s", name);
+    for (size_t i = 0; i < data_size; i++)
+    {
+        TRACE_ENCLAVE("[%ld]-0x%02X", i, data[i]);
+    }
+    TRACE_ENCLAVE("\n");
+}
+
+// Compute the sha256 hash of given data.
+int ecall_dispatcher::Sha256(
+    const uint8_t* data,
+    size_t data_size,
+    uint8_t sha256[32])
+{
+    int ret = -1;
+    unsigned int sha_length_in_bytes = 32;
+    EVP_MD_CTX* mdctx;
+
+    if ((mdctx = EVP_MD_CTX_new()) == NULL)
+    {
+        TRACE_ENCLAVE("Message digest context instatntiation failed");
+        goto exit;
+    }
+
+    if (1 != EVP_DigestInit_ex(mdctx, EVP_sha256(), NULL))
+    {
+        TRACE_ENCLAVE(
+            "Message digest context initialization to SHA-256 failed");
+        goto exit;
+    }
+
+    if (1 != EVP_DigestUpdate(mdctx, data, data_size))
+    {
+        TRACE_ENCLAVE("Message digest context update to input data failed");
+        goto exit;
+    }
+
+    if (1 != EVP_DigestFinal_ex(mdctx, sha256, &sha_length_in_bytes))
+    {
+        TRACE_ENCLAVE("Message Digest Final failed");
+        goto exit;
+    }
+
+    ret = 0;
+    EVP_MD_CTX_free(mdctx);
```

Comment:
Shouldn't this be done after exit: ?

Reply:
Yes. Will make the change in next commit.